src/TopSurface.cpp

```diff
@@ -37,14 +37,28 @@ bool TopSurface::ironing(const SliceMeshStorage& mesh, const GCodePathConfig& li
     const bool zig_zaggify_infill = pattern == EFillMethod::ZIG_ZAG;
     constexpr bool connect_polygons = false; // midway connections can make the surface less smooth
     const coord_t line_spacing = mesh.settings.get<coord_t>("ironing_line_spacing");
-    const coord_t outline_offset = -mesh.settings.get<coord_t>("ironing_inset");
     const coord_t line_width = line_config.getLineWidth();
     const std::vector<AngleDegrees>& top_most_skin_angles = (mesh.settings.get<size_t>("roofing_layer_count") > 0) ? mesh.roofing_angles : mesh.skin_angles;
     assert(top_most_skin_angles.size() > 0);
     const AngleDegrees direction = top_most_skin_angles[layer.getLayerNr() % top_most_skin_angles.size()] + AngleDegrees(90.0); //Always perpendicular to the skin lines.
     constexpr coord_t infill_overlap = 0;
     constexpr int infill_multiplier = 1;
     constexpr coord_t shift = 0;
+
+    coord_t ironing_inset = -mesh.settings.get<coord_t>("ironing_inset");
+    if (pattern == EFillMethod::ZIG_ZAG && ironing_inset == 0)
+    {
+        //Compensate for the outline_offset decrease that takes place when using the infill generator to generate ironing with the zigzag pattern
+        const float width_scale = (float)mesh.settings.get<coord_t>("layer_height") / mesh.settings.get<coord_t>("infill_sparse_thickness");
+        ironing_inset += width_scale * line_width / 2;
+    }
+    else if (pattern == EFillMethod::CONCENTRIC)
+    {
+        //Counteract the outline_offset increase that takes place when using the infill generator to generate ironing with the concentric pattern
+        ironing_inset -= line_width / 2;
```

Comment:
This inset also doesn't seem to be working properly. I get this g-code with a 1cm cube centred around the origin, towards the very end:

```
G0 F7200 X4.9 Y4.9
G1 F1200 X-4.9 Y4.9 E111.96965
G1 X-4.9 Y-4.9 E111.97579
G1 X4.9 Y-4.9 E111.98194
G1 X4.9 Y4.9 E111.98808
;TIME_ELAPSED:670.436642
```
This is the last and outermost contour of ironing. It's offset from the original shape by 0.1mm here, which is way more than the `0.4mm / 2 * 10% = 0.02mm` that I'd expect.

To reproduce, use the same project file as mentioned before, but change the ironing pattern to concentric.

Reply:
Fixed in https://github.com/Ultimaker/CuraEngine/pull/1251/commits/b49ddf7c5c73c76c5a6aac788cb8acafeef60794 by moving the ironing line by ironing_line_width/2 inwards.